api/versions.go

```diff
@@ -270,3 +274,63 @@ func (self *version2_0) HandleRequest(requestType string, request []string, m ma
 		return self.baseVersion.HandleRequest(requestType, request, m, w, r)
 	}
 }
+
+// API v1.3
+
+type version1_3 struct {
+	baseVersion *version1_2
+}
+
+// v1.3 builds on v1.2.
+func newVersion1_3(v *version1_2) *version1_3 {
+	return &version1_3{
+		baseVersion: v,
+	}
+}
+
+func (self *version1_3) Version() string {
+	return "v1.3"
+}
+
+func (self *version1_3) SupportedRequestTypes() []string {
+	return append(self.baseVersion.SupportedRequestTypes(), eventsApi)
+}
+
+func (self *version1_3) HandleRequest(requestType string, request []string, m manager.Manager, w http.ResponseWriter, r *http.Request) error {
+	switch {
+	case requestType == eventsApi:
+		query, eventsFromAllTime, err := getEventRequest(request)
+		if err != nil {
+			return err
+		}
+		glog.V(2).Infof("Api - Events(%v)", query)
+
+		if eventsFromAllTime {
+			allEvents, err := m.GetPastEvents(query)
+			if err != nil {
+				return err
+			}
+			return writeResult(allEvents, w)
+		} else {
+			// every time URL is entered to watch, a channel is created here
+			eventChannel := make(chan *events.Event)
+			err = m.WatchForEvents(query, eventChannel)
+
+			go func() {
```

Comment:
No need to make this a separate Go routine, one is spun up per HTTP request so it is okay to block it.